Wrong underline length in numpy doc strings

<!--Describe your proposed enhancement in detail.-->

Some headings in the doc strings do not have the right number of underlining character and this causes extra warning in the doc build.

This does not fail the build though.

For example: https://github.com/nilearn/nilearn/actions/runs/4122627440/jobs/7119633978#step:33:323

### note

Probably in the category
- low effort
- low impact

Can probably be tackled at the same time as #2528

List of lines to fix:

```bash
$ flake8 --select D409 nilearn

nilearn/_utils/data_gen.py:500:1: D409 Section underline should match the length of its name
nilearn/_utils/ndimage.py:15:1: D409 Section underline should match the length of its name
nilearn/_utils/ndimage.py:15:1: D409 Section underline should match the length of its name
nilearn/_utils/niimg.py:107:1: D409 Section underline should match the length of its name
nilearn/_utils/niimg.py:107:1: D409 Section underline should match the length of its name
nilearn/connectome/group_sparse_cov.py:28:1: D409 Section underline should match the length of its name
nilearn/datasets/func.py:423:1: D409 Section underline should match the length of its name
nilearn/datasets/func.py:771:1: D409 Section underline should match the length of its name
nilearn/datasets/func.py:771:1: D409 Section underline should match the length of its name
nilearn/datasets/func.py:813:1: D409 Section underline should match the length of its name
nilearn/datasets/func.py:813:1: D409 Section underline should match the length of its name
nilearn/datasets/neurovault.py:1151:1: D409 Section underline should match the length of its name
nilearn/datasets/neurovault.py:1151:1: D409 Section underline should match the length of its name
nilearn/datasets/neurovault.py:2616:1: D409 Section underline should match the length of its name
nilearn/datasets/neurovault.py:2616:1: D409 Section underline should match the length of its name
nilearn/datasets/neurovault.py:2656:1: D409 Section underline should match the length of its name
nilearn/datasets/neurovault.py:2656:1: D409 Section underline should match the length of its name
nilearn/datasets/utils.py:820:1: D409 Section underline should match the length of its name
nilearn/image/image.py:58:1: D409 Section underline should match the length of its name
nilearn/image/resampling.py:192:1: D409 Section underline should match the length of its name
nilearn/image/resampling.py:703:1: D409 Section underline should match the length of its name
nilearn/interfaces/fmriprep/load_confounds.py:81:1: D409 Section underline should match the length of its name
nilearn/interfaces/fmriprep/load_confounds_strategy.py:56:1: D409 Section underline should match the length of its name
nilearn/masking.py:656:1: D409 Section underline should match the length of its name
nilearn/masking.py:656:1: D409 Section underline should match the length of its name
nilearn/plotting/displays/_slicers.py:284:1: D409 Section underline should match the length of its name
nilearn/plotting/edge_detect.py:35:1: D409 Section underline should match the length of its name
nilearn/plotting/edge_detect.py:120:1: D409 Section underline should match the length of its name
nilearn/plotting/find_cuts.py:35:1: D409 Section underline should match the length of its name
nilearn/plotting/img_plotting.py:1077:1: D409 Section underline should match the length of its name
nilearn/surface/surface.py:275:1: D409 Section underline should match the length of its name
nilearn/surface/surface.py:796:1: D409 Section underline should match the length of its name
nilearn/surface/surface.py:882:1: D409 Section underline should match the length of its name
```

--- a/nilearn/_utils/data_gen.py
+++ b/nilearn/_utils/data_gen.py
@@ -530,7 +530,7 @@ def write_fake_fmri_data_and_design(shapes,
         A list of paths to the generated design matrix files.
 
     See Also
-    ------
+    --------
     nilearn._utils.data_gen.generate_fake_fmri_data_and_design
 
     """

--- a/nilearn/_utils/ndimage.py
+++ b/nilearn/_utils/ndimage.py
@@ -15,12 +15,12 @@ def largest_connected_component(volume):
     """Return the largest connected component of a 3D array.
 
     Parameters
-    -----------
+    ----------
     volume : numpy.ndarray
         3D boolean array indicating a volume.
 
     Returns
-    --------
+    -------
     volume : numpy.ndarray
         3D boolean array with only one connected component.
 

--- a/nilearn/_utils/niimg.py
+++ b/nilearn/_utils/niimg.py
@@ -106,8 +106,8 @@ def _get_target_dtype(dtype, target_dtype):
 def load_niimg(niimg, dtype=None):
     """Load a niimg, check if it is a nibabel SpatialImage and cast if needed
 
-    Parameters:
-    -----------
+    Parameters
+    ----------
 
     niimg: Niimg-like object
         See https://nilearn.github.io/stable/manipulating_images/input_output.html  # noqa:E501
@@ -118,8 +118,8 @@ def load_niimg(niimg, dtype=None):
         data will be converted to int32 if dtype is discrete and float32 if it
         is continuous.
 
-    Returns:
-    --------
+    Returns
+    -------
     img: image
         A loaded image object.
     """

--- a/nilearn/connectome/group_sparse_cov.py
+++ b/nilearn/connectome/group_sparse_cov.py
@@ -56,7 +56,7 @@ def compute_alpha_max(emp_covs, n_samples):
         dense matrix.
 
     References
-    ---------
+    ----------
     .. footbibliography::
 
     """

--- a/nilearn/datasets/func.py
+++ b/nilearn/datasets/func.py
@@ -787,7 +787,7 @@ def fetch_localizer_calculation_task(n_subjects=1, data_dir=None, url=None,
         'cmaps': string list, giving paths to nifti contrast maps
 
     Notes
-    ------
+    -----
     This function is only a caller for the fetch_localizer_contrasts in order
     to simplify examples reading and understanding.
     The 'calculation (auditory and visual cue)' contrast is used.
@@ -829,7 +829,7 @@ def fetch_localizer_button_task(data_dir=None, url=None,
         - 'anat': string, giving paths to normalized anatomical image
 
     Notes
-    ------
+    -----
     This function is only a caller for the fetch_localizer_contrasts in order
     to simplify examples reading and understanding.
     The 'left vs right button press' contrast is used.

--- a/nilearn/datasets/func.py
+++ b/nilearn/datasets/func.py
@@ -553,7 +553,7 @@ def fetch_localizer_contrasts(contrasts, n_subjects=None, get_tmaps=False,
     .. footbibliography::
 
     See Also
-    ---------
+    --------
     nilearn.datasets.fetch_localizer_calculation_task
     nilearn.datasets.fetch_localizer_button_task
 
@@ -793,7 +793,7 @@ def fetch_localizer_calculation_task(n_subjects=1, data_dir=None, url=None,
     The 'calculation (auditory and visual cue)' contrast is used.
 
     See Also
-    ---------
+    --------
     nilearn.datasets.fetch_localizer_button_task
     nilearn.datasets.fetch_localizer_contrasts
 
@@ -835,7 +835,7 @@ def fetch_localizer_button_task(data_dir=None, url=None,
     The 'left vs right button press' contrast is used.
 
     See Also
-    ---------
+    --------
     nilearn.datasets.fetch_localizer_calculation_task
     nilearn.datasets.fetch_localizer_contrasts
 

--- a/nilearn/datasets/neurovault.py
+++ b/nilearn/datasets/neurovault.py
@@ -1154,8 +1154,8 @@ def neurosynth_words_vectorized(word_files, verbose=3, **kwargs):
     dictionaries for each image, this function reads it and returns a
     vocabulary list and a term weight matrix.
 
-    Parameters:
-    -----------
+    Parameters
+    ----------
     word_files : Container
         The paths to the files from which to read word weights (each
         is supposed to contain the Neurosynth response for a
@@ -1168,8 +1168,8 @@ def neurosynth_words_vectorized(word_files, verbose=3, **kwargs):
     Keyword arguments are passed on to
     ``sklearn.feature_extraction.DictVectorizer``.
 
-    Returns:
-    --------
+    Returns
+    -------
     frequencies : numpy.ndarray
         An (n images, vocabulary size) array. Each row corresponds to
         an image, and each column corresponds to a word. The words are
@@ -2636,12 +2636,12 @@ def fetch_neurovault_motor_task(data_dir=None, verbose=1):
             - 'description', a short description of the Neurovault dataset.
 
     Notes
-    ------
+    -----
     The 'left vs right button press' contrast is used:
     https://neurovault.org/images/10426/
 
     See Also
-    ---------
+    --------
     nilearn.datasets.fetch_neurovault_ids
     nilearn.datasets.fetch_neurovault
     nilearn.datasets.fetch_neurovault_auditory_computation_task
@@ -2677,12 +2677,12 @@ def fetch_neurovault_auditory_computation_task(data_dir=None, verbose=1):
             - 'description', a short description of the Neurovault dataset.
 
     Notes
-    ------
+    -----
     The 'auditory_calculation_vs_baseline' contrast is used:
     https://neurovault.org/images/32980/
 
     See Also
-    ---------
+    --------
     nilearn.datasets.fetch_neurovault_ids
     nilearn.datasets.fetch_neurovault
     nilearn.datasets.fetch_neurovault_motor_task

--- a/nilearn/datasets/utils.py
+++ b/nilearn/datasets/utils.py
@@ -819,8 +819,8 @@ def _fetch_files(data_dir, files, resume=True, verbose=1, session=None):
 def _tree(path, pattern=None, dictionary=False):
     """Return a directory tree under the form of a dictionaries and list
 
-    Parameters:
-    -----------
+    Parameters
+    ----------
     path : string
         Path browsed.
 

--- a/nilearn/image/image.py
+++ b/nilearn/image/image.py
@@ -87,7 +87,7 @@ def high_variance_confounds(imgs, n_confounds=5, percentile=2.,
             Highest variance confounds. Shape: *(number_of_scans, n_confounds)*.
 
         Notes
-        ------
+        -----
         This method is related to what has been published in the literature
         as 'CompCor' (Behzadi NeuroImage 2007).
 
@@ -99,7 +99,7 @@ def high_variance_confounds(imgs, n_confounds=5, percentile=2.,
         - Returns a given number (n_confounds) of signals from the SVD with
           highest singular values.
 
-        See also
+        See Also
         --------
         nilearn.signal.high_variance_confounds
 

--- a/nilearn/image/resampling.py
+++ b/nilearn/image/resampling.py
@@ -199,7 +199,7 @@ def get_mask_bounds(img):
             background.
 
         Returns
-        --------
+        -------
         xmin, xmax, ymin, ymax, zmin, zmax : floats
             The world-space bounds (field of view) occupied by the
             non-zero values in the image
@@ -706,7 +706,7 @@ def reorder_img(img, resample=None):
     'ValueError' is raised.
 
     Parameters
-    -----------
+    ----------
     img : Niimg-like object
         See :ref:`extracting_data`.
         Image to reorder.

--- a/nilearn/interfaces/fmriprep/load_confounds.py
+++ b/nilearn/interfaces/fmriprep/load_confounds.py
@@ -261,7 +261,7 @@ def load_confounds(img_files,
     :func:`nilearn.interfaces.fmriprep.load_confounds_strategy`
 
     References
-    -----------
+    ----------
     .. footbibliography::
 
     """

--- a/nilearn/interfaces/fmriprep/load_confounds_strategy.py
+++ b/nilearn/interfaces/fmriprep/load_confounds_strategy.py
@@ -189,7 +189,7 @@ def load_confounds_strategy(img_files, denoise_strategy="simple", **kwargs):
     :func:`nilearn.interfaces.fmriprep.load_confounds`
 
     References
-    -----------
+    ----------
     .. footbibliography::
 
     """

--- a/nilearn/masking.py
+++ b/nilearn/masking.py
@@ -658,7 +658,7 @@ def apply_mask(imgs, mask_img, dtype='f',
     Read the time series from the given Niimg-like object, using the mask.
 
     Parameters
-    -----------
+    ----------
     imgs : :obj:`list` of 4D Niimg-like objects
         See :ref:`extracting_data`.
         Images to be masked. list of lists of 3D images are also accepted.
@@ -683,7 +683,7 @@ def apply_mask(imgs, mask_img, dtype='f',
         Default=True.
 
     Returns
-    --------
+    -------
     session_series : :class:`numpy.ndarray`
         2D array of series with shape (image number, :term:`voxel` number)
 

--- a/nilearn/plotting/displays/_slicers.py
+++ b/nilearn/plotting/displays/_slicers.py
@@ -284,7 +284,7 @@ class BaseSlicer:
         """ Contour a 3D map in all the views.
 
         Parameters
-        -----------
+        ----------
         %(img)s
             Provides image to plot.
 

--- a/nilearn/plotting/edge_detect.py
+++ b/nilearn/plotting/edge_detect.py
@@ -50,7 +50,7 @@ def _edge_detect(image, high_threshold=.75, low_threshold=.4):
         Default=0.4.
 
     Returns
-    --------
+    -------
     grad_mag : 2D array of floats
         The magnitude of the gradient.
 
@@ -125,7 +125,7 @@ def _edge_map(image):
         The image that the edges are extracted from.
 
     Returns
-    --------
+    -------
     edge_mask : 2D masked array
         A mask of the edge as a masked array with parts without
         edges masked and the large extents detected with lower

--- a/nilearn/plotting/find_cuts.py
+++ b/nilearn/plotting/find_cuts.py
@@ -35,7 +35,7 @@ def find_xyz_cut_coords(img, mask_img=None, activation_threshold=None):
     """ Find the center of the largest activation connected component.
 
     Parameters
-    -----------
+    ----------
     img : 3D Nifti1Image
         The brain map.
 

--- a/nilearn/plotting/img_plotting.py
+++ b/nilearn/plotting/img_plotting.py
@@ -1143,7 +1143,7 @@ def plot_connectome(adjacency_matrix, node_coords,
         Default=False.
 
     See Also
-    ---------
+    --------
     nilearn.plotting.find_parcellation_cut_coords : Extraction of node
         coords on brain parcellations.
     nilearn.plotting.find_probabilistic_atlas_cut_coords : Extraction of

--- a/nilearn/surface/surface.py
+++ b/nilearn/surface/surface.py
@@ -274,8 +274,8 @@ def _sample_locations(mesh, affine, radius, kind='auto', n_points=None,
 def _masked_indices(sample_locations, img_shape, mask=None):
     """Get the indices of sample points which should be ignored.
 
-    Parameters:
-    -----------
+    Parameters
+    ----------
     sample_locations : array, shape(n_sample_locations, 3)
         The coordinates of candidate interpolation points.
 
@@ -805,7 +805,7 @@ def load_surf_mesh(surf_mesh):
         Mesh object with "coordinates" and "faces" attributes.
 
     Returns
-    --------
+    -------
     mesh : Mesh
         With the fields "coordinates" and "faces", each containing a
         numpy.ndarray
@@ -900,7 +900,7 @@ def load_surface(surface):
                     or (n_time_points, n_vertices)
 
     Returns
-    --------
+    -------
     surface : Surface
         With the fields "mesh" (Mesh object) and "data" (numpy.ndarray).
 

